lib/meta_inspector/parsers/images.rb

```diff
@@ -10,12 +12,44 @@ def images
         self
       end
 
+      # Returns either the Facebook Open Graph image, twitter suggested image or
+      # the first image in the image collection
+      def best
+        owner_suggested || images_collection.first
+      end
+
       # Returns the parsed image from Facebook's open graph property tags
       # Most major websites now define this property and is usually relevant
       # See doc at http://developers.facebook.com/docs/opengraph/
       # If none found, tries with Twitter image
-      def best
-        meta['og:image'] || meta['twitter:image'] || images_collection.first
+      def owner_suggested
+        meta['og:image'] || meta['twitter:image']
+      end
+
+      # Returns the largest image from the image collection,
+      # filtered for images that are more square than 10:1 or 1:10
+      def largest(download_images = true)
```

Comment:
I think it's clearer if we use named arguments for the options, so instead of `page.images.largest(true)` we have `page.images.largest(download_images: true`).

But, I see this as a global option of the Images parser, I mean, if we change #best to return the largest image when there is no owner suggested image, we should be able to set this download option as well here.

What I would do is passing this option from Document, like the rest of the options, so the user can set it from the top level:

```ruby
page = MetaInspector.new('http://example.com', download_images: false)

class Document
  def initialize
     options          = defaults.merge(options)
     ...
     @download_images = options[:download_images]
     ...
     @parser = MetaInspector::Parser.new(self, exception_log:   @exception_log,
                                               download_images: @download_images)
  end

  private

  def defaults
    { ...
      :download_images => true }
  end
end

class Parser
  def initialize(document, options = {})
    ...
    @download_images = options[:download_images]
    ...
    @images_parser = MetaInspector::ImagesParser.new(self, download_images: @download_images)
  end
end

class ImagesParser
  def initialize(options = {})
    ...
    @download_images = options[:download_images]
  end

  def largest
    ...
    if @download_images ...
  end
end
```